pulsar-broker/src/main/java/org/apache/pulsar/broker/service/PulsarChannelInitializer.java

```diff
@@ -38,6 +48,11 @@
     private final NettySslContextBuilder sslCtxRefresher;
     private final ServiceConfiguration brokerConf;
 
+    private final Cache<SocketAddress, ServerCnx> connections = Caffeine.newBuilder()
```

Comment:
nit: Why not Cache<ServerCnx, ServerCnx> and avoid possibly allocating the SocketAddress? We don't care about the keys in any case. Maybe add a comment that the you're using Cache to avoid having to clean stuff up.

Reply:
The socket address is already kept in the `ServerCnx` itself, so it will not be a new allocation here.

Reply:
Added comment